### PR TITLE
Split Precommit Integration Tests into a separate Build

### PR DESCRIPTION
**Pull request by tgroh**
Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Make sure there is a [JIRA issue](https://issues.apache.org/jira/projects/BEAM/issues/) filed for the change (usually before you start working on it).  Trivial changes like typos do not require a JIRA issue.  Your pull request should address just this issue, without pulling in other changes.
 - [ ] Each commit in the pull request should have a meaningful subject line and body.
 - [ ] Format the pull request title like `[BEAM-XXX] Fixes bug in ApproximateQuantiles`, where you replace `BEAM-XXX` with the appropriate JIRA issue.
 - [ ] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
 - [ ] Run `mvn clean verify` to make sure basic checks pass. A more thorough check will be performed on your pull request automatically.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

---
Integration tests consume a significant amount of time, and are easily
separable from the remaining tests.

### CODE REVIEW

**Comment by tgroh** [comment]
For your consideration.

R: @jasonkuster @kennknowles @lukecwik

File: .test-infra/jenkins/job_beam_PreCommit_Java_Example_Integration.groovy [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+    60)
+
+  // Set Maven parameters.
+  common_job_properties.setMavenConfig(delegate)
+
+  // Sets that this is a PreCommit job which runs the integration-tests.
+  common_job_properties.setPreCommit(delegate, 'mvn failsafe:integration-test -pl examples/java -am', 'Run Java Integration PreCommit')
+
+  // Maven goals for this job: The Java Examples and their dependencies
+  goals('''\
```
Comment (kennknowles) [inline .test-infra/jenkins/job_beam_PreCommit_Java_Example_Integration.groovy]: Jason made the good point that if you build an array of args and then join them on spaces, you avoid the possibility of omitting a backslash, which I have done in ~every one of my changes. Yours looks already though :-)

File: .test-infra/jenkins/job_beam_PreCommit_Java_Example_Integration.groovy [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+    --projects examples/java \
+    --also-make \
+    -D checkstyle.skip \
+    -D findbugs.skip \
+    -D pullRequest=$ghprbPullId \
+    help:effective-settings \
+    clean \
+    compile \
+    test-compile \
+    failsafe:integration-test \
```
Comment (kennknowles) [inline .test-infra/jenkins/job_beam_PreCommit_Java_Example_Integration.groovy]: Notably, invoking `integration-test` will neglect the cleanup (I don't know if we have any). I think if you add `post-integration-test` you are OK, but maybe just raw `verify`?

It is covered pretty early here: http://maven.apache.org/surefire/maven-failsafe-plugin/

File: .test-infra/jenkins/job_beam_PreCommit_Java_MavenInstall.groovy [hunk 3]
```diff
@@ -44,7 +44,7 @@ mavenJob('beam_PreCommit_Java_MavenInstall') {
   goals('''\
     --batch-mode \
     --errors \
-    --activate-profiles release,jenkins-precommit,direct-runner,dataflow-runner,spark-runner,flink-runner,apex-runner \
+    --activate-profiles release,direct-runner,dataflow-runner,spark-runner,flink-runner,apex-runner \
```
Comment (kennknowles) [inline .test-infra/jenkins/job_beam_PreCommit_Java_MavenInstall.groovy]: These profiles are only for the examples to run against different runners. I think there should be either none or just `direct-runner`.

**Comment by reuvenlax** [comment]
Do we still need this?

**Comment by tgroh** [comment]
I don't think it's currently high-priority. Closing.

**Pull request closed without merging**